### PR TITLE
Eagerly build availability index

### PR DESCRIPTION
**Pull request by sung-luxuryescapes**
POST `/api/search/index-availability` starts a process to eagerly refresh availability indices of all existing room types by calling reservation service.

depends on https://github.com/brandsExclusive/svc-reservation/pull/673

### CODE REVIEW

File: src/controllers/scriptController.test.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  beforeEach(async () => {
+    await redis.flushdb();
+  });
+
+  afterEach(async () => {
+    reservationStub.restore();
+    await redis.flushdb();
+  });
+
+  it.only("indexes the available room_type counts", async () => {
```
Comment (joshuapaling) [inline src/controllers/scriptController.test.ts:44]: Remove `.only`. Would be nice if we had a lint rule that checked for this. I know `svc-ee-offer` does - there's a plugin called `eslint-plugin-no-only-tests`

**Comment by joshuapaling** [comment]
@sung-luxuryescapes an example of the appoach I mentioned is the recent stuff I did on Google autocomplete proxy. Here's the controller test, which is very minimal: https://github.com/brandsExclusive/fn-search/blob/master/src/controllers/googlePlaceController.test.ts

But that's because the logic in the controller itself is also very minimal - it just calls another module and responds with the results: https://github.com/brandsExclusive/fn-search/blob/master/src/controllers/googlePlaceController.ts

All the business logic, and the tests for that logic, are self-contained in https://github.com/brandsExclusive/fn-search/tree/master/src/lib/augmentPlaceResults

**Comment by joshuapaling** [comment]
Side comment: another anti-pattern we should be careful to avoid as we move towards this pattern, which I've seen happen in past jobs, is when someone adds in the detailed unit tests, but KEEPS the detailed controller tests too. At that point, the controller tests are redundant, test the same thing a second time, and just make for bloated test suites that are a nightmare to maintain.

**Comment by sung-luxuryescapes** [comment]
@joshuapaling good idea. Fixed by https://github.com/brandsExclusive/fn-search/pull/82/commits/0aff34321c4d4a798fc787a7553bdab93abbd61c

**Pull request merged**